ios/brave-ios/Tests/DataTests/DomainTests.swift

```diff
@@ -241,3 +238,22 @@ class DomainTests: CoreDataTestCase {
     XCTAssertFalse(raydiumDomain.walletPermissions(for: .sol, account: walletSolAccount))
   }
 }
+
+extension Domain {
+  fileprivate class func performChangesOnDomain(
```

Comment:
Why is this function necessary? It's only used in Unit Tests.

Reply:
This is only for tests we need to ensure changes are saved before running tests. There was already somethiing like this but it was specific to changing one of the BraveShield enums. 

Since I removed `adblockAndTp` from that enum, I needed to create another method to do this so I created a more general one and privatized it to this file. 

Actually that existing function was only used in this test, so I can remove it and put it here:

```
extension Domain {
  private func setBraveShield(
    shield: BraveShield,
    isOn: Bool?,
    context: NSManagedObjectContext
  ) {

    let setting = (isOn == shield.globalPreference ? nil : isOn) as NSNumber?
    switch shield {
    case .allOff: shield_allOff = setting
    case .fpProtection: shield_fpProtection = setting
    case .noScript: shield_noScript = setting
    }
  }
  
  fileprivate class func setBraveShield(
    forUrl url: URL,
    shield: BraveShield,
    isOn: Bool?,
    isPrivateBrowsing: Bool
  ) {
    let _context: WriteContext = isPrivateBrowsing ? .new(inMemory: true) : .new(inMemory: false)
    setBraveShieldInternal(forUrl: url, shield: shield, isOn: isOn, context: _context)
  }
  
  fileprivate class func setBraveShieldInternal(
    forUrl url: URL,
    shield: BraveShield,
    isOn: Bool?,
    context: WriteContext = .new(inMemory: false)
  ) {
    DataController.perform(context: context) { context in
      // Not saving here, save happens in `perform` method.
      let domain = Domain.getOrCreateInternal(
        url,
        context: context,
        saveStrategy: .delayedPersistentStore
      )
      domain.setBraveShield(shield: shield, isOn: isOn, context: context)
    }
  }
  
  fileprivate class func performChangesOnDomain(
    for url: URL,
    isPrivateBrowsing: Bool,
    changes: @escaping (Domain) -> Void
  ) {
    let context = WriteContext.new(inMemory: isPrivateBrowsing)
    DataController.perform(context: context) { context in
      // Not saving here, save happens in `perform` method.
      let domain = Domain.getOrCreateInternal(
        url,
        context: context,
        saveStrategy: .delayedPersistentStore
      )
      changes(domain)
    }
  }
}
```

And of course, I can slim this down and remove some of the duplicated unecessary stuff, but all the functions in the above (except for the last one) are existing functions which were only used in these tests and I wanted to show this.

Reply:
If you would like to see a commit i can do that. but in the end these are the changes:

<img width="999" alt="Screenshot 2024-07-09 at 4 48 53 PM" src="https://github.com/brave/brave-core/assets/909331/5925cbc8-ce1f-4336-bb35-0bc019459b4b">
<img width="1006" alt="Screenshot 2024-07-09 at 4 49 05 PM" src="https://github.com/brave/brave-core/assets/909331/7dc021ca-64f9-474e-8e8b-426b7f5a7e17">